tensorflow_serving/batching/batching_session.h

```diff
@@ -99,6 +99,23 @@ struct BatchingSessionOptions {
   //
   // If left empty, no rounding/padding is performed.
   std::vector<int> allowed_batch_sizes;
+
+  // If set to true, padding is performed for tensors of the same name
+  // but with unequal dimensions, so that all tensors of the same name have
+  // equal dim sizes.
+  //
+  // For example:
```

Comment:
What will happen if a user tries to batch together tensor with different dimension sizes, and they have set pad_variable_length_inputs=false? They'll get an error Status right (and not a crash)? Please document this, and also ensure there's a unit test (perhaps there already is one).